src/json-ld.net/Core/DocumentLoader.cs

```diff
@@ -6,31 +6,73 @@
 {
     public class DocumentLoader
     {
+        const int MAX_REDIRECTS = 20;
+
+        /// <summary>An HTTP Accept header that prefers JSONLD.</summary>
+        /// <remarks>An HTTP Accept header that prefers JSONLD.</remarks>
+        public const string AcceptHeader = "application/ld+json, application/json;q=0.9, application/javascript;q=0.5, text/javascript;q=0.5, text/plain;q=0.2, */*;q=0.1";
+
+        /// <exception cref="JsonLDNet.Core.JsonLdError"></exception>
+        public RemoteDocument LoadDocument(string url)
+        {
+            return LoadDocumentAsync(url).GetAwaiter().GetResult();
+        }
+
         /// <exception cref="JsonLDNet.Core.JsonLdError"></exception>
-        public virtual RemoteDocument LoadDocument(string url)
+        public async Task<RemoteDocument> LoadDocumentAsync(string url)
         {
-#if !PORTABLE && !IS_CORECLR
             RemoteDocument doc = new RemoteDocument(url, null);
-            HttpWebResponse resp;
 
             try
             {
-                HttpWebRequest req = (HttpWebRequest)HttpWebRequest.Create(url);
-                req.Accept = AcceptHeader;
-                resp = (HttpWebResponse)req.GetResponse();
-                bool isJsonld = resp.Headers[HttpResponseHeader.ContentType] == "application/ld+json";
-                if (!resp.Headers[HttpResponseHeader.ContentType].Contains("json"))
+                HttpResponseMessage httpResponseMessage;
+
+                int redirects = 0;
+                int code;
+                string redirectedUrl = url;
+
+                // Manually follow redirects because .NET Core refuses to auto-follow HTTPS->HTTP redirects.
+                do
+                {
+                    HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, redirectedUrl);
+                    httpRequestMessage.Headers.Add("Accept", AcceptHeader);
+                    httpResponseMessage = await JSONUtils._HttpClient.SendAsync(httpRequestMessage);
+                    if (httpResponseMessage.Headers.TryGetValues("Location", out var location))
+                    {
+                        redirectedUrl = location.First();
+                    }
+
+                    code = (int)httpResponseMessage.StatusCode;
+                } while (redirects++ < MAX_REDIRECTS && code >= 300 && code < 400);
+
+                if (redirects >= MAX_REDIRECTS)
+                {
+                    throw new JsonLdError(JsonLdError.Error.LoadingDocumentFailed, $"Too many redirects - {url}");
+                }
+
+                if (code >= 400)
+                {
+                    throw new JsonLdError(JsonLdError.Error.LoadingDocumentFailed, $"HTTP {code} {url}");
+                }
+
+                bool isJsonld = httpResponseMessage.Content.Headers.ContentType.MediaType == "application/ld+json";
+
+                // From RFC 6839, it looks like we should accept application/json and any MediaType ending in "+json".
```

Comment:
Issue #12 was complaining that this check is too strict, but it looks like it has been more liberal than the reported behavior since the beginning of time. I actually made it a little stricter... It was `.Contains("json")`, which seems wrong. I checked some RFCs, and it looks like `"application/json"` and anything ending with the suffix `+json` should be accepted here.